pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -329,6 +329,12 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 
 		availablePhysicalCPUs := p.GetAvailablePhysicalCPUs(s).Size()
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+				cpuAllocatedQuantity := cs.AllocatedResources[v1.ResourceCPU]
+				availablePhysicalCPUs += int(cpuAllocatedQuantity.Value())
```

Comment:
`GetContainerStatus` will return the resources which are requested for the pod and not yet granted?
How do we make sure the the delta is an integral amount of cpus (e.g. 2, but not 3.35?)

Reply:
Thanks for your valuable feedback, your comment with non integral amount of cpus was an eye opener for me. I was performing my tests with one container, it doesn't work the patch need to improve.

Reply:
> `GetContainerStatus` will return the resources which are requested for the pod and not yet granted? 

If a pod exists and a pod resource resize is attempted then GetContainerStatus returns the last kubelet agreed resources. allocatedCpuQuantity holds the 'truth' of the from state.

> How do we make sure the the delta is an integral amount of cpus (e.g. 2, but not 3.35?)

It is handled in https://github.com/kubernetes/kubernetes/blob/657c2f9f793d64405e4a3fddd440b6d70dd70256/pkg/kubelet/cm/cpumanager/policy_static.go#L324-L353